packages/compiler/core/binder.ts

```diff
@@ -477,21 +492,27 @@ export function createBinder(program: Program): Binder {
         return declareScriptMember(node, flags, name);
       default:
         const key = name ?? node.id.sv;
-        const symbol = createSymbol(node, key, flags, scope.symbol);
+        const symbol = createSymbol(node, key, flags, scope?.symbol);
         mutate(node).symbol = symbol;
         mutate(scope.locals!).set(key, symbol);
         return symbol;
     }
   }
 
+  function bindSymbol(node: Node, flags: SymbolFlags): Sym {
+    const symbol = createSymbol(node, "-", flags, scope?.symbol);
+    mutate(node).symbol = symbol;
+    return symbol;
+  }
+
   function declareNamespaceMember(node: Declaration, flags: SymbolFlags, name?: string) {
     if (
       flags & SymbolFlags.Namespace &&
       mergeNamespaceDeclarations(node as NamespaceStatementNode, scope)
     ) {
       return;
     }
-    const key = name ?? node.id.sv;
+    const key = name ?? node.id?.sv;
```

Comment:
How do we handle the now possibly undefined key?   (is it really possibly undefined)?